packages/tools/index.ts

```diff
@@ -0,0 +1,52 @@
+import asc from "assemblyscript/bin/asc";
+import fs from "fs";
+import { BaseTool } from "./assembly/base";
+
+type ToolInfo = {
+  id: string;
+  name: string;
+};
+
+export interface IToolFactory {
+  getToolList(): ToolInfo[];
+  downloadTool(toolId: string, storageDir?: string): void;
+  loadTool(filePath: string): Promise<BaseTool>;
```

Comment:
good idea to have a factory! How to simplify it's usage for the caller?
```suggestion
  getToolList(): ToolInfo[];
  loadTool(toolId: string): Promise<BaseTool>;
```

Reply:
but as a user of the tools package, I would probably prefer to directly use the tool (the factory is still good for enumeration and dynamically creating a tool), this means we need a wrapper for each tool that I can use like this:

`new WikipediaTool()`